cfgov/v1/admin_views.py

```diff
@@ -17,14 +22,35 @@ def cdn_is_configured():
 
 
 def purge(url=None):
+    akamai_config = settings.WAGTAILFRONTENDCACHE.get('akamai', {})
+    cloudfront_config = settings.WAGTAILFRONTENDCACHE.get(
+        'cloudfront', {})
+
     if url:
-        purge_url_from_cache(url)
-        return "flushed %s" % url
+        # Use the Wagtail frontendcache PurgeBatch to perform the purge
+        batch = PurgeBatch()
+        batch.add_url(url)
+
+        # If the URL matches any of our cloudfront distributions, invalidate
+        # with that backend
+        if any(k for k in cloudfront_config.get('DISTRIBUTION_ID', {})
+               if k in url):
+            logger.info('Purging {} from cloudfront'.format(url))
+            batch.purge(backends='cloudfront')
+
+        # Otherwise invalidate with our default backend
+        else:
+            logger.info('Purging {} from akamai'.format(url))
+            batch.purge(backends='akamai')
```

Comment:
Shouldn't `backends` be a list, per [this comment](https://github.com/wagtail/wagtail/blob/v1.13.4/wagtail/contrib/wagtailfrontendcache/utils.py#L145)?

Reply:
Listified in 570be79